install-guard.sh

```diff
@@ -56,10 +58,29 @@ get_os_type() {
 	esac
 }
 
+get_version() {
+	# Get the version from the -v option, if provided.
+	while getopts 'v:' OPTION; do
+		case "$OPTION" in
+			v)
+				VERSION="$OPTARG"
```

Comment:
```suggestion
				VERSION="$OPTARG"
				# Check if the version is valid.
				if ! [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
					err "Invalid version '${VERSION}'. Version must in semver format, ie. '1.2.3'."
				fi
				;;

```
Could include a simple regex version check here.  Current omitted for brevity.